libpod/info.go

```diff
@@ -93,10 +93,25 @@ func (r *Runtime) hostInfo() (*define.HostInfo, error) {
 		return nil, errors.Wrapf(err, "error getting Seccomp profile path")
 	}
 
+	// CGroups version
+	unified, err := cgroups.IsCgroup2UnifiedMode()
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading cgroups mode")
+	}
+
+	// Get Map of all available controllers
+	availableControllers, err := cgroups.GetAvailableControllers(nil, unified)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting available cgroup controllers")
+	}
+
 	info := define.HostInfo{
 		Arch:           runtime.GOARCH,
 		BuildahVersion: buildah.Version,
 		CgroupManager:  r.config.Engine.CgroupManager,
+		MemoryLimit:    availableControllers["memory"],
```

Comment:
Wondering it may make more sense to just return `CgroupControllers []string` here.

(Docker-compatible REST API should emulate Docker REST API, though)

Reply:
@AkihiroSuda `docker info` spits boolean for separate controllers so i tried keeping it same. But i guess a common string for all available controllers would be better. WDYT ?

Reply:
Yes, for Podman CLI, returning the common string slice SGTM.
Docker-compatible REST API should retain Docker-compatible booleans.

Reply:
@AkihiroSuda I tried this but dumping all the available controllers is making `info` too noisy, for instance on my machine even single liner has too much content. I guess docker only prints standard controllers maybe because this is too verbose but i am cool with this.
On my machine `info` shows this
```bash
cgroupControllers:
  - blkio
  - cpu
  - cpu,cpuacct
  - cpuacct
  - cpuset
  - devices
  - freezer
  - hugetlb
  - memory
  - net_cls
  - net_cls,net_prio
  - net_prio
  - perf_event
  - pids
  - rdma
  - systemd
  - unified
```

Reply:
For cgroup v1, the right way is to parse /proc/cgroups

Reply:
@AkihiroSuda Would still give the same output as i have all controllers enabled on v1 
```bash
$ cat /proc/cgroups
#subsys_name	hierarchy	num_cgroups	enabled
cpuset	0	230	1
cpu	0	230	1
cpuacct	0	230	1
blkio	0	230	1
memory	0	230	1
devices	0	230	1
freezer	0	230	1
net_cls	0	230	1
perf_event	0	230	1
net_prio	0	230	1
hugetlb	0	230	1
pids	0	230	1
rdma	0	230	1
```

Reply:
not "same" :)

Reply:
@AkihiroSuda ah i get it. So i guess logic for existing `getAvailableControllers` https://github.com/containers/podman/blob/master/pkg/cgroups/cgroups.go#L129 for cgroupv1 needs to be changed or i can add a new function but dont think redundant code for similar task is a good idea. @rhatdan @giuseppe @AkihiroSuda  If we can mutually agree upon having `CgroupControllers []string` in `info` then i think we can proceed with removing old logic in `master` and use `/proc/cgroups` for cgroupv1

Reply:
cgroupv1 is not really supported for rootless, even if manually chowned as the delegation is unsafe.

For root, we can parse `/proc/cgroups` or `/proc/self/cgroups` but for rootless we can just force every controller to be disabled.

Reply:
@giuseppe @AkihiroSuda Thank a lot . I have added changes, this is resolved in latest commit